pom.xml

```diff
@@ -138,12 +140,22 @@
       <dependency>
         <groupId>org.apache.maven</groupId>
         <artifactId>maven-embedder</artifactId>
-        <version>3.8.4</version>
+        <version>${maven.version}</version>
       </dependency>
       <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-utils</artifactId>
-        <version>3.4.1</version>
+        <version>3.0.20</version><!-- must match version exported from Maven: https://github.com/apache/maven/blob/cab6659f9874fa96462afef40fcf6bc033d58c1c/pom.xml#L52 -->
```

Comment:
Would it make sense to import the Maven POM instead of declaring versions for each of these? I can imagine them falling out of sync in the future.

Reply:
The Maven POM contains much more than just the used API and would restrict the versions of dependencies not exported by the Core Classloader. Unfortunately a Maven BOM is only available since Maven 4 (https://issues.apache.org/jira/browse/MNG-6806), so I don't see a way to prevent this manual management of versions.

Reply:
would be better to use last version of p-u which can include security fixes and perf inprovment.
this doesn't have to be the same. and p-u is really backward compat

Reply:
@olamy This code only uses the package always exported through the Maven Core class loader from p-u

Reply:
@kwin I think the key point here is that plexus-utils is backwards compatible. You should be able to mix newer versions of the dependency with older versions of Maven.

Reply:
IMHO this would require forwards-compatibility. Newer version of p-u might introduce new methods in `org.codehaus.plexus.util.xml.Xpp3Dom` which might not be implemented in older versions (this would still mean newer versions are backwards compatible). If this code now uses this new methods, it builds fine, but breaks at run time when running in Maven using an older version of p-u. Therefore it is much safer to depend on the older version one claims to be compatible with (not the newest one)!

Reply:
@kwin What prevents us from rolling back to older versions of this dependency if and when it causes problems for you in the future? For now we are tackling a theoretical problem.

Reply:
The primary goal of dependencies are that
a) things compile
b) make sure that the code can run in a runtime where the dependency in this version (or a newer one for those which claim to be backward-compatible) is given at run time

If you depend on a newer version a) would still work, but b) no longer necessarily. So the question boils down to what is more important:

i) silencing false positives of security scanners (as the dependency is in fact provided)
ii) ensuring that you don't accidentally break backwards compatibility.

For me it is ii).
But I think I am done now with all discussions. Do whatever you like with this PR.